menu/menu_displaylist.c

```diff
@@ -3706,19 +3744,51 @@ unsigned menu_displaylist_build_list(file_list_t *list, enum menu_displaylist_ct
             }
          }
          break;
+      case DISPLAYLIST_DROPDOWN_LIST_PLAYLIST_LABEL_DISPLAY_MODE:
+         {
+            playlist_t *playlist             = playlist_get_cached();
+
+            if (playlist)
+            {
+               enum playlist_label_display_mode label_display_mode = playlist_get_label_display_mode(playlist);
+               int i;
+
+               for (i = 0; MENU_ENUM_LABEL_VALUE_PLAYLIST_MANAGER_LABEL_DISPLAY_MODE_DEFAULT + i != MENU_ENUM_LABEL_VALUE_PLAYLIST_MANAGER_LABEL_DISPLAY_MODE_KEEP_REGION_AND_DISC_INDEX; i += 3)
```

Comment:
As above, this `i += 3` stuff is not safe. You'll need a well-defined `playlist_label_display_mode` <-> `msg_hash_enums` conversion.

(sorry - I know this is annoying!)